tests/cypress/e2e/model-spec.cy.js

```diff
@@ -0,0 +1,128 @@
+describe('DiAGRAM model creation spec', () => {
+
+    beforeEach(() => {
+        cy.visit('/model.html')
+    })
+
+    it('should have all sections needed for model creation', () => {
+        cy.get('.box-body #questions-section #questions .question .question-header h3').then($sectionHeaders => {
+            expect($sectionHeaders[0]).to.contain.text('Digital Object');
+            expect($sectionHeaders[1]).to.contain.text('Storage Medium');
+            expect($sectionHeaders[2]).to.contain.text('Replication and Refreshment');
+            expect($sectionHeaders[3]).to.contain.text('Replication and Refreshment');
+            expect($sectionHeaders[4]).to.contain.text('Operating Environment');
+            expect($sectionHeaders[5]).to.contain.text('Operating Environment');
+            expect($sectionHeaders[6]).to.contain.text('Physical Disaster');
+            expect($sectionHeaders[7]).to.contain.text('Checksum');
+            expect($sectionHeaders[8]).to.contain.text('System Security');
+            expect($sectionHeaders[9]).to.contain.text('System Security');
+            expect($sectionHeaders[10]).to.contain.text('System Security');
+            expect($sectionHeaders[11]).to.contain.text('System Security');
+            expect($sectionHeaders[12]).to.contain.text('Information Management');
+            expect($sectionHeaders[13]).to.contain.text('Information Management');
+            expect($sectionHeaders[14]).to.contain.text('Information Management');
+            expect($sectionHeaders[15]).to.contain.text('Technical Skills');
+        });
+    })
+
+    it('should click through the model creation pages with default values and produce a model at the end', () => {
+        cy.get('#models-section-container').should('be.not.visible')
+        cy.get('.box-body #model-start').click({force: true});
+
+        // Name container, give name to the model
+        cy.get('.box-body #name-container #model-name').should('be.visible')
+        cy.get('.box-body #name-container #model-name').type('Test Model')
+        cy.get('.box-body #prequestions #name-container .button-row .btn-next').should('be.visible')
+        cy.get('.box-body #prequestions #name-container .button-row .btn-next').click({force: true})
+
+        // Name captured in the questions-ancillary section
+        cy.get('.box-body #questions-section #questions-ancillary .questions-ancillary-top-row .questions-name-container p span').then($modelName => {
+            expect($modelName[0]).to.contain.text('Test Model');
+        });
+
+        // Model type container
+        assertHiddenAndVisibleToggled('sm-q1', 'do-q1');
+        cy.get('.box-body #questions-section #questions-buttons .button-row .btn-next').click({force: true})
```

Comment:
Since this is repeated another 15 times, could this be added to a function and called?

Reply:
That would be replacing one line with a one line function call, so I had deliberately kept it out of the function